plugins/haconfig/keepalived/generator.go

```diff
@@ -0,0 +1,172 @@
+package keepalived
+
+import (
+	"fmt"
+	"strconv"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	kclientcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	"github.com/golang/glog"
+
+	dapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+	"github.com/openshift/origin/pkg/haconfig"
+)
+
+const defaultInterface = "eth0"
+const libModulesVolumeName = "lib-modules"
+const libModulesPath = "/lib/modules"
+
+//  Get kube client configuration from a file containing credentials for
+//  connecting to the master.
+func getClientConfig(path string) *kclient.Config {
+	if 0 == len(path) {
+		glog.Fatalf("You must specify a .kubeconfig file path containing credentials for connecting to the master with --credentials")
+	}
+
+	rules := &kclientcmd.ClientConfigLoadingRules{ExplicitPath: path, Precedence: []string{}}
+	credentials, err := rules.Load()
+	if err != nil {
+		glog.Fatalf("Could not load credentials from %q: %v", path, err)
+	}
+
+	config, err := kclientcmd.NewDefaultClientConfig(*credentials, &kclientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		glog.Fatalf("Credentials %q error: %v", path, err)
+	}
+
+	if err := kclient.LoadTLSFiles(config); err != nil {
+		glog.Fatalf("Unable to load certificate info using credentials from %q: %v", path, err)
+	}
+
+	return config
+}
+
+//  Generate the HA failover monitor (keepalived) container environment entries.
+func generateEnvEntries(name string, options *haconfig.HAConfigCmdOptions, kconfig *kclient.Config) app.Environment {
+	watchPort := strconv.Itoa(options.WatchPort)
+	replicas := strconv.Itoa(options.Replicas)
+	insecureStr := strconv.FormatBool(kconfig.Insecure)
+	unicastStr := strconv.FormatBool(options.UseUnicast)
+
+	return app.Environment{
+		"OPENSHIFT_MASTER":    kconfig.Host,
+		"OPENSHIFT_CA_DATA":   string(kconfig.CAData),
+		"OPENSHIFT_KEY_DATA":  string(kconfig.KeyData),
+		"OPENSHIFT_CERT_DATA": string(kconfig.CertData),
+		"OPENSHIFT_INSECURE":  insecureStr,
+
+		"OPENSHIFT_HA_CONFIG_NAME":       name,
+		"OPENSHIFT_HA_VIRTUAL_IPS":       options.VirtualIPs,
+		"OPENSHIFT_HA_NETWORK_INTERFACE": options.NetworkInterface,
+		"OPENSHIFT_HA_MONITOR_PORT":      watchPort,
+		"OPENSHIFT_HA_REPLICA_COUNT":     replicas,
+		"OPENSHIFT_HA_USE_UNICAST":       unicastStr,
+		// "OPENSHIFT_HA_UNICAST_PEERS":     "127.0.0.1",
+	}
+}
+
+//  Generate the HA failover monitor (keepalived) container configuration.
+func generateFailoverMonitorContainerConfig(name string, options *haconfig.HAConfigCmdOptions, env app.Environment) *kapi.Container {
+	containerName := fmt.Sprintf("%s-%s", name, options.Type)
+
+	imageName := fmt.Sprintf("%s-%s", options.Type, haconfig.DefaultName)
+	image := options.ImageTemplate.ExpandOrDie(imageName)
+
+	ports := make([]kapi.ContainerPort, 1)
+	ports[0] = kapi.ContainerPort{
+		ContainerPort: options.ServicePort,
```

Comment:
Is this keepaliveds interconnect port?  Is that their default assigned port?  If so, add a comment to that effect